### PR TITLE
docs: add better documentation to setup project

### PR DESCRIPTION
**Pull request by Yimura**
I still need to go over each and every step to make sure I didn't skip anything.

Technically it should work...

Read the SETUP guide here:
https://github.com/YimMenu/YimMenu/blob/documentation/docs/SETUP.md

### CODE REVIEW

File: docs/SETUP.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+## Requirements
+
+ - [Winget](#winget)
+ - [Git](#git)
+ - [C++ Desktop Development](#c-desktop-development)
+ - [Cloning YimMenu](#cloning-yimmenu)
+ - [Choose your preferred IDE](#choose-your-preferred-ide)
+    - [Visual Studio](#visual-studio)
+    - [Visual Studio Code](#visual-studio-code)
+    - [No IDE or CLI]()
```
Comment (tupoy-ya) [inline docs/SETUP.md]: No compiling in the table of contents.

File: docs/SETUP.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Open Visual Studio Code and open the local folder to which you cloned `YimMenu`.
+Visual Studio Code should then prompt you to generate the CMake project, allow it to generate the project this might take a while depending on the performance of your computer.
+
+After that has finished you should be good to go and start compiling YimMenu.
+
+At the bottom of your IDE you should have the following things:
+![VS Code Build Options](img/vsc-build.png)
+
+Try match your IDE with what you see in the above screenshot after which you can press the Build button.
```
Comment (tupoy-ya) [inline docs/SETUP.md]: Maybe i'm blind but there is no instruction on how to press `F7` in VSCode.

**Review by tupoy-ya — DISMISSED** [review]
After 2 years in development finally the "Clear and well structured source code" is removed.

File: docs/SETUP.md (hunk 2 above)
Comment (Yimura) [reply docs/SETUP.md]: Wasn't aware this was a thing, will add it.

**Comment by GeopJr** [comment]
Instead of using the readme, I'd vote for moving dev docs to the wiki https://github.com/YimMenu/YimMenu/wiki

otherwise, have at least the images *outside* the repo as it will increase its size by a lot and is irreversible unless you fpush them out (imgur?)

**Comment by Yimura** [comment]
I guess Imgur is a possibility also, however a wiki would be nice too.

Just don't want to spend ages to write the articles.

**Comment by Seanghost117** [comment]
> Instead of using the readme, I'd vote for moving dev docs to the wiki https://github.com/YimMenu/YimMenu/wiki
> 
> otherwise, have at least the images _outside_ the repo as it will increase its size by a lot and is irreversible unless you fpush them out (imgur?)

agreed

**Comment by Yimura** [comment]
I checked in a clean Windows 11 VM and the guide appears to work perfectly.

**Comment by shoaib842** [comment]
> [](https://github.com/YimMenu/YimMenu/blob/documentation/docs/SETUP.md)

hey there i have followed down your setup.md guidance by having the latest visual studio and curl installation setting up it in path variable, but getting down this error 
image

Need help to solve this out.

**Comment by shoaib842** [comment]
> I still need to go over each and every step to make sure I didn't skip anything.
> 
> Technically it should work...
> 
> Read the SETUP guide here: https://github.com/YimMenu/YimMenu/blob/documentation/docs/SETUP.md

hey there i have followed down your setup.md guidance by having the latest visual studio and curl installation setting up it in path variable, but getting down this error
image

Need help to solve this out

**Review by xiaoxiao921 — DISMISSED** [review]
Any reason this is not merged? much better than what we have currently

**Comment by Yimura** [comment]
> Any reason this is not merged? much better than what we have currently

People are looking towards using the Wiki feature rather than storing the images in the source which would bloat the checking out of code.

**Pull request merged**